### PR TITLE
Generated html reports for e2e tests and screenshots for test failure

### PR DESCRIPTION
**Pull request by monika-Dianomic**
## What do these changes do?

Generated html reports for e2e tests and screenshots for test failure

- html result and screenshots store in `tmp` folder

## Are there changes in behavior for the user?

No

## Related issue number

None

## Checklist

- [x] Code is well written w.r.t. best practices and proper comments

### CODE REVIEW

File: protractor.conf.js [hunk 1]
```diff
@@ -17,12 +17,63 @@ exports.config = {
...
     showColors: true,
     defaultTimeoutInterval: 30000,
-    print: function() {}
+    print: function () { }
   },
   onPrepare() {
+    var jasmineReporters = require('jasmine-reporters');
+    jasmine.getEnv().addReporter(new jasmineReporters.JUnitXmlReporter({
+      consolidateAll: true,
+      savePath: 'testresults',
```
Comment (mshariq-nerd) [inline protractor.conf.js]: rename `testresults` to `test_results`

File: protractor.conf.js [hunk 2]
```diff
@@ -17,12 +17,63 @@ exports.config = {
...
+      filePrefix: 'xmloutput'
```
Comment (mshariq-nerd) [inline protractor.conf.js]: rename `xmloutput` to  `xml_output`

File: package.json [hunk 3]
```diff
@@ -26,11 +26,13 @@
     "angular2-multiselect-dropdown": "~2.7.0",
     "chart.js": "~2.7.2",
     "core-js": "~2.5.6",
+    "jasmine-reporters": "~2.3.1",
     "lodash-es": "~4.17.10",
     "moment": "~2.22.1",
     "ng-sidebar": "~6.0.4",
     "ngx-mask": "~1.0.3",
     "ngx-progressbar": "~2.1.1",
+    "protractor-html-reporter": "~1.3.2",
```
Comment (praveen-garg) [inline package.json]: looks this is not actively maintained!

File: protractor.conf.js [hunk 4]
```diff
@@ -20,60 +20,17 @@ exports.config = {
...
-    jasmine.getEnv().addReporter(new jasmineReporters.JUnitXmlReporter({
-      consolidateAll: true,
-      savePath: 'testresults',
-      filePrefix: 'xmloutput'
-    }));
+    var HtmlReporter = require('protractor-beautiful-reporter');
+    // Add a screenshot reporter and store screenshots to `/tmp/screenshots`:
+      jasmine.getEnv().addReporter(new HtmlReporter({
+         baseDirectory: 'tmp/screenshots',
+         jsonsSubfolder: 'jsons',
```
Comment (praveen-garg) [inline protractor.conf.js]: jsons?

**Comment by praveen-garg** [comment]
How to use it? and see reports?

**Comment by monika-Dianomic** [comment]
After running the e2e tests, test results generated under `tmp/screenshots` folder. If you open `report.html` `({base_url}/foglamp-gui/tmp/screenshots/report.html)` in the browser, it shows Test result with total pass/fail/error and screenshots attached with failed tests.

test_result

File: package.json [hunk 5]
```diff
@@ -31,6 +31,7 @@
     "ng-sidebar": "~6.0.4",
     "ngx-mask": "~1.0.3",
     "ngx-progressbar": "~2.1.1",
+    "protractor-beautiful-reporter": "~1.2.1",
```
Comment (praveen-garg) [inline package.json]: this should be dev dep!

File: protractor.conf.js [hunk 6]
```diff
@@ -17,9 +17,17 @@ exports.config = {
...
     showColors: true,
     defaultTimeoutInterval: 30000,
-    print: function() {}
+    print: function () { }
   },
   onPrepare() {
+    var HtmlReporter = require('protractor-beautiful-reporter');
+    // Add a screenshot reporter and store screenshots to `/tmp/screenshots`:
+      jasmine.getEnv().addReporter(new HtmlReporter({
+         baseDirectory: 'tmp/screenshots',
```
Comment (praveen-garg) [inline protractor.conf.js]: this storing everything to screenshots!

File: protractor.conf.js [hunk 7]
```diff
@@ -17,9 +17,19 @@ exports.config = {
...
     showColors: true,
     defaultTimeoutInterval: 30000,
-    print: function() {}
+    print: function () { }
   },
   onPrepare() {
+    var HtmlReporter = require('protractor-beautiful-reporter');
+    // Add a screenshot reporter and store screenshots to `/tmp`:
+      jasmine.getEnv().addReporter(new HtmlReporter({
+         baseDirectory: 'tmp',
```
Comment (mshariq-nerd) [inline protractor.conf.js]: baseDirectory name should be more appropriate like test-reports

Comment (praveen-garg) [reply protractor.conf.js]: if its inside e2e then we can  simply  name it `report` else `e2e-test-report`

**Pull request merged**